frontend/src/app/main/ui/auth/recovery_request.cljs

```diff
@@ -22,15 +22,21 @@
 (s/def ::recovery-request-form (s/keys :req-un [::email]))
 
 (mf/defc recovery-form
-  []
+  [{:keys [on-success-callback] :as props}]
   (let [form      (fm/use-form :spec ::recovery-request-form :initial {})
         submitted (mf/use-state false)
 
+        default-success-finish #(st/emit! (dm/info (tr "auth.notifications.recovery-token-sent")))
+
+        success-finish #(if (nil? on-success-callback)
+                   (default-success-finish)
+                   (on-success-callback (:email %)))
```

Comment:
I think that this anonymos function can be inlined directly in the on-success callback, it is a single use anonymous function of 2 lines.